File: eco-core/core-plugin/src/main/java/com/willfp/stattrackers/commands/CommandStatTrackers.java
package com.willfp.stattrackers.commands;

import com.willfp.eco.core.EcoPlugin;
import com.willfp.eco.core.command.CommandHandler;
import com.willfp.eco.core.command.impl.PluginCommand;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class CommandStatTrackers extends PluginCommand {
    /**
     * Instantiate a new command handler.
     *
     * @param plugin The plugin for the commands to listen for.
     */
    public CommandStatTrackers(@NotNull final EcoPlugin plugin) {
        super(plugin, "stattrackers", "stattrackers.command.stattrackers", false);

        this.addSubcommand(new CommandReload(plugin))
                .addSubcommand(new CommandGive(plugin))
                .addSubcommand(new CommandActivestat(plugin));
    }

    @Override
    public void onExecute(@NotNull CommandSender sender, @NotNull List<String> args) {
        sender.sendMessage(this.getPlugin().getLangYml().getMessage("invalid-command"));
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/stattrackers/commands/CommandStatTrackers.kt
package com.willfp.stattrackers.commands

import com.willfp.eco.core.command.impl.PluginCommand
import com.willfp.stattrackers.plugin
import com.willfp.stattrackers.stats.StatTrackersGUI
import com.willfp.stattrackers.stats.canTrackStats
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

object CommandStatTrackers : PluginCommand(
    plugin,
    "stattrackers",
    "stattrackers.command.stattrackers",
    true
) {
    init {
        this.addSubcommand(CommandReload)
            .addSubcommand(CommandGive)
    }

    override fun onExecute(sender: CommandSender, args: List<String>) {
        val player = sender as Player

        val item = player.inventory.itemInMainHand

        if (!item.canTrackStats) {
            player.sendMessage(plugin.langYml.getMessage("item-cannot-have-trackers"))
            return
        }

        StatTrackersGUI.open(player)
    }
}
